### ISSUE TITLE
Computing SSM on 2 volumes in the same file

### ISSUE THREAD

**Issue by FrancoisMay** [SCIInstitute/ShapeWorks#1394]
I tried to use ShapeWorks on two volumes in the same file. ShapeWorks selected part of the volumes, but I did not figured how it did it. Might it be possible to add a way to compute SSM on multiple bones in the same file please?

**Comment by sheryjoe** [comment]
> I tried to use ShapeWorks on two volumes in the same file. 
@FrancoisMay are these mult-label volumes?  i.e., each bone has a different label (voxel value) in the same volume and same label across all volumes in the study?

**Comment by FrancoisMay** [comment]
This is not my dataset, and I do not know how to find this information, but I attached you an example of the file I used. 
[file.zip](https://github.com/SCIInstitute/ShapeWorks/files/6824723/file.zip)

**Comment by akenmorris** [comment]
@sheryjoe , it's multiple meshes within an STL file:

<img width="537" alt="image" src="https://user-images.githubusercontent.com/1693349/125824776-a21d514a-dbe2-4d99-987b-7831e69ade40.png">

**Comment by sheryjoe** [comment]
@akenmorris @FrancoisMay if Studio would accept this type of input, there should be labels added to the mesh files to indicate the domains and labels should be consistent across samples. @akenmorris thoughts?

**Comment by akenmorris** [comment]
What kind of labeling?  I’m not really aware of labelling within mesh files.

**Comment by sheryjoe** [comment]
@akenmorris a constant scalar field per bone? however, I am not sure if there are existing tools that can provide such labeling unless we provide such labeling as a grooming step. Here the user would have to select the same bone across all samples.  another way is to separate connected components (assuming each bone has its own component), the user should label which component belongs to which domain. In all cases, the user has to define the label for each domain.

**Comment by akenmorris** [comment]
I agree this makes sense.  However, do we need a threshold for number of triangles to consider a component?  We have an isolate groom step primarily because many meshes come in with stray triangles (e.g. additional meshes) off to the side.

**Comment by sheryjoe** [comment]
@akenmorris One option is to have the user indicate the expected number of domains (N) and to select the top largest N connected components.
